Source/JavaScriptCore/dfg/DFGAbstractInterpreterInlines.h

```diff
@@ -2218,51 +2222,81 @@ bool AbstractInterpreter<AbstractStateType>::executeEffects(unsigned clobberLimi
                             setConstant(node, jsBoolean(result == TriState::True));
                             didFold = true;
                             break;
                         }
                         break;
                     case Array::Int32Array:
-                        if (auto result = foldComparisonWithRange(node->op(), INT32_MIN, INT32_MAX, rightValue); result != TriState::Indeterminate) {
+                        if (auto result = foldComparisonWithRange<int64_t>(node->op(), INT32_MIN, INT32_MAX, rightValue); result != TriState::Indeterminate) {
                             setConstant(node, jsBoolean(result == TriState::True));
                             didFold = true;
                             break;
                         }
                         break;
                     case Array::Uint32Array:
-                        if (auto result = foldComparisonWithRange(node->op(), 0, UINT32_MAX, rightValue); result != TriState::Indeterminate) {
+                        if (auto result = foldComparisonWithRange<int64_t>(node->op(), 0, UINT32_MAX, rightValue); result != TriState::Indeterminate) {
+                            setConstant(node, jsBoolean(result == TriState::True));
+                            didFold = true;
+                            break;
+                        }
+                        break;
+                    default:
+                        break;
+                    }
+                    if (didFold)
+                        break;
+                }
+                if (rightConst.isNumber()) {
+                    double rightValue = rightConst.asNumber();
+                    switch (arrayMode.type()) {
+                    case Array::Float16Array:
+                        if (auto result = foldComparisonWithRange<double>(node->op(), Float16::min(), Float16::max(), rightValue); result != TriState::Indeterminate) {
+                            setConstant(node, jsBoolean(result == TriState::True));
+                            didFold = true;
+                            break;
+                        }
+                        break;
+                    case Array::Float32Array:
+                        if (auto result = foldComparisonWithRange<double>(node->op(), std::numeric_limits<float>::min(), std::numeric_limits<float>::max(), rightValue); result != TriState::Indeterminate) {
+                            setConstant(node, jsBoolean(result == TriState::True));
+                            didFold = true;
+                            break;
+                        }
+                        break;
+                    case Array::Float64Array:
+                        if (auto result = foldComparisonWithRange<double>(node->op(), std::numeric_limits<double>::min(), std::numeric_limits<double>::max(), rightValue); result != TriState::Indeterminate) {
```

Comment:
Float16 / float / double's min and max values are -infinity and +infinity. Also it is possible that these typed arrays have NaN. So I don't think we can fold comparisons. Can you explain how they work with them?

Reply:
Oooops, I forget about Infinity, as you said probably we cannot fold comparisons in this way...